bundles/core/org.eclipse.smarthome.core.thing/src/main/java/org/eclipse/smarthome/core/thing/setup/ThingSetupManager.java

```diff
@@ -600,43 +549,11 @@ private Thing addThing(ThingTypeUID thingTypeUID, ThingUID thingUID, Configurati
                 thing.setProperty(key, properties.get(key));
             }
         }
-
-        String itemName = toItemName(thing.getUID());
-        GroupItem groupItem = new GroupItem(itemName);
-        groupItem.addTag(TAG_THING);
-        groupItem.setLabel(label);
-        groupItem.addGroupNames(groupNames);
         addThingSafely(thing);
-        addItemSafely(groupItem);
-        addItemThingLinkSafely(new ItemThingLink(itemName, thing.getUID()));
-
-        ThingType thingType = thingTypeRegistry.getThingType(thingTypeUID);
-        if (thingType != null) {
-            List<ChannelGroupDefinition> channelGroupDefinitions = thingType.getChannelGroupDefinitions();
-            for (ChannelGroupDefinition channelGroupDefinition : channelGroupDefinitions) {
-                GroupItem channelGroupItem = new GroupItem(
-                        getChannelGroupItemName(itemName, channelGroupDefinition.getId()));
-                channelGroupItem.addTag(TAG_CHANNEL_GROUP);
-                channelGroupItem.addGroupName(itemName);
-                addItemSafely(channelGroupItem);
-            }
+        addItems(label, groupNames, thing, thingTypeUID);
```

Comment:
I just learned that addItems actually only adds the group item for the thing (and its groups), but not the items for the channels (which is done through enableChannels). I wonder if we should hence find a better name for the method? Something like "createGroupItems"?